tests/component/test_component.py

```diff
@@ -442,27 +447,23 @@ def test_pandas_transform_component(metadata):
         "False",
     ]
 
+    init_called = 0
+
     class MyPandasComponent(PandasTransformComponent):
         def __init__(self, *, flag, value):
             assert flag == "success"
             assert value == 1
+            nonlocal init_called
+            init_called += 1
 
         def transform(self, dataframe):
             assert isinstance(dataframe, pd.DataFrame)
             return dataframe.rename(columns={"images": "embeddings"})
 
-    init = patch_method_class(MyPandasComponent.__init__)
-    transform = patch_method_class(MyPandasComponent.transform)
     executor_factory = ExecutorFactory(MyPandasComponent)
     executor = executor_factory.get_executor()
-    with mock.patch.object(MyPandasComponent, "__init__", init), mock.patch.object(
-        MyPandasComponent,
-        "transform",
-        transform,
-    ):
```

Comment:
This was giving issues with recursion when pickling. I couldn't completely figure out the cause and how to fix it.